config/fi_provider.m4

```diff
@@ -123,9 +123,10 @@ AC_DEFUN([FI_PROVIDER_SETUP],[
 	# If this provider was requested for direct build, ensure that
 	# provider's fi_direct.h exists in tree. Error otherwise.
 	AS_IF([test x"$enable_direct" = x"$1"],
-		[AC_MSG_CHECKING(for $srcdir/prov/$1/include/rdma/fi_direct.h)
-		 AS_IF([test -f "$srcdir/prov/$1/include/rdma/fi_direct.h"],
-			[AC_MSG_RESULT(yes)],
+		[AC_MSG_CHECKING(for prov/$1/include/rdma/fi_direct.h)
+		 AS_IF([test -f "prov/$1/include/rdma/fi_direct.h"],
+			[AC_MSG_RESULT(yes)
```

Comment:
I don't remember removing $srcdir from the path, that may be an unintentional mistake.  Overwriting CFLAGS below is not desired either.  If CFLAGS is going to be used to define the global we need to append " -DFABRIC_DIRECT_ENABLED" to the end of CFLAGS.